bin.src/plot_photoCalib.py

```diff
@@ -0,0 +1,264 @@
+
+    bbox = image.getBBox()
+    extent = (0, bbox.getWidth()*step, 0, image.getHeight()*step)
+
+    plt.figure()
+    percentiles = np.nanpercentile(image.getArray(), [percentile, 100-percentile])
+    ax = plt.imshow(image.getArray(), cmap=colormap, origin='lower',
+                    extent=extent, vmin=percentiles[0], vmax=percentiles[1])
+    plt.title(title)
+    plt.xlabel("x focal plane pixels")
+    plt.ylabel("y focal plane pixels")
+    colorbar(ax)
+    plt.tight_layout()
+
+    plt.savefig(filename)
+    print('Wrote', filename)
+
+
+def main():
+    import argparse
+    parser = argparse.ArgumentParser(description=__doc__,
+                                     formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument("repo", metavar="repo", help="Directory containing butler repository.")
+    parser.add_argument("--step", default=8, type=int,
+                        help="Steps between samples across the focal plane.")
+    parser.add_argument("--tract", default=0, type=int,
+                        help="Tract to plot (until gen3 butler can manage this automatically).")
+    parser.add_argument("--percentile", type=float, default=1,
+                        help="Show the percentile/100-percentile range of the photoCalib.")
+    parser.add_argument("-v", "--verbose", action="store_true",
+                        help="Print extra information while producing plots.")
+    parser.add_argument("--colormap", default="magma",
+                        help="matplotlib.Colormap name to use when generating the 2d plot.")
+    args = parser.parse_args()
+
+    butler = lsst.daf.persistence.Butler(inputs=args.repo)
+    dataIds = getValidDataIds(butler, args.tract)
+    visits = np.unique([dataId['visit'] for dataId in dataIds])
+    chips = np.unique([dataId['ccd'] for dataId in dataIds])
+    if args.verbose:
+        print("Found these visits:", visits)
+        print("And these chips:", chips)
+
+    for visit in visits:
+        if args.verbose:
+            print()
+            print("Processing visit", visit)
+        plotVisitPhotoCalib(visit, butler, tract=args.tract, chips=chips,
+                            step=args.step, percentile=args.percentile,
+                            chipScaling=False,
+                            colormap=args.colormap, verbose=args.verbose)
+        plotVisitPhotoCalib(visit, butler, tract=args.tract, chips=chips,
+                            step=args.step, percentile=args.percentile,
+                            chipScaling=True, meanCalib=True,
+                            colormap=args.colormap, verbose=args.verbose)
+        plotVisitPhotoCalib(visit, butler, tract=args.tract, chips=chips,
+                            step=args.step, percentile=args.percentile,
+                            chipScaling=True, singleCalib=True,
+                            colormap=args.colormap, verbose=args.verbose)
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
Block of code looks good assuming it outputs the plots you are expecting. One ask is to put numpydoc strings on all of the methods. From the looks of it this is only about 3 or so.